src/core/network.js

```diff
@@ -115,20 +133,17 @@ var NetworkManager = (function NetworkManagerClosure() {
         pendingRequest.expectedStatus = 200;
       }
 
-      if (args.onProgressiveData) {
-        // Some legacy browsers might throw an exception.
-        try {
-          xhr.responseType = 'moz-chunked-arraybuffer';
-        } catch(e) {}
-        if (xhr.responseType === 'moz-chunked-arraybuffer') {
-          pendingRequest.onProgressiveData = args.onProgressiveData;
-          pendingRequest.mozChunked = true;
-        } else {
-          xhr.responseType = 'arraybuffer';
-        }
+//#if (GENERIC || FIREFOX || MOZCENTRAL)
+      if (args.onProgressiveData && supportsMozChunked) {
+        xhr.responseType = 'moz-chunked-arraybuffer';
+        pendingRequest.onProgressiveData = args.onProgressiveData;
+        pendingRequest.mozChunked = true;
       } else {
         xhr.responseType = 'arraybuffer';
       }
+//#else
```

Comment:
>  and because we don't want preprocessor magic to eliminate the boolean,

Au contraire, I want to eliminate the boolean for FF/CHROMIUM, means that in non-FF/CHROMIUM code (which will be uncommented by default) the supportsMozChunked feature test and check will be present. For FF/CHROMIUM cases, we use preprocessor to provide optimized versions of the same code (which will be commented) with assumption of supportsMozChunked=true/false.

Does it make more sense?

In the future, I hope we can re-write this code and define better classes/interfaces than XHR that will be more suitable for our operations.